### PR TITLE
Add support for Sidekiq 7.0

### PR DESCRIPTION
**Pull request by jeremycw**
Sidekiq 7.0 now depends on the redis-client gem instead of the redis gem. This caused issues with `sidekiq-client.rb` and `sidekiq-worker.rb`

### CODE REVIEW

**Comment by Ferenc-** [comment]
Hi @jeremycw,
Thank you for your interest in contributing!

I would like to kindly ask you to sign your commit with `git commit -s`,
indicating that you as the submitter accept the [Developer's Certificate of Origin 1.1](https://elinux.org/Developer_Certificate_Of_Origin),
and submit this work under our [MIT License](https://github.com/instana/ruby-sensor/blob/master/LICENSE) 
which grants the usual MIT License rights `to use, copy, modify, merge, publish, distribute, sublicense, and/or sell` to anyone.

File: lib/instana/instrumentation/sidekiq-client.rb [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
   module Instrumentation
     class SidekiqClient
+      def self.redis_url
+        Sidekiq.redis_pool.with do |client|
+          host, port = 
+            case
+            when client.respond_to?(:config)
+              [client.config.host, client.config.port]
+            when client.respond_to?(:connection)
+              [client.connection[:host], client.connection[:host]]
```
Comment (Ferenc-) [inline lib/instana/instrumentation/sidekiq-client.rb:14]: This line appear to assign the host to both, `host` and `port`.
Is it possible that you meant the following?
```suggestion
              [client.connection[:host], client.connection[:port]]
```

**Review by Ferenc- — CHANGES_REQUESTED** [review]
So in this commit support for Sidekiq 7.0, seems to be added, but tests remain with `gem "sidekiq", ">= 5.0"`, "< 6.0" and `gem "sidekiq", "~> 4.2.10"`.
So I am afraid, that we would never to know when this support breaks or if it ever really worked. Do you think it would be possible to change that?

And have you perhaps considered moving this functionality into `instrumentation/redis.rb`?

**Comment by jeremycw** [comment]
Hey @Ferenc- I've made some updates based on your suggestions. I've left this functionality where it is for now since in it's current incarnation it's very specific to Sidekiq requiring access to the client through `Sidekiq.redis_pool` and the `client` object in this case is actually an instance of `Sidekiq::RedisClientAdapter::CompatClient`

**Comment by Ferenc-** [comment]
Thank you for making the changes.
In `test/instrumentation/sidekiq-client_test.rb` we have a number of test cases.
When running those on this PR branch with Sidekiq 7.0 the following appear to be failing :

1. test_enqueue_failure
2. test_successful_worker_starts_new_trace
3. test_failed_worker_starts_new_trace
4. test_failed_worker_continues_previous_trace
5. test_successful_worker_continues_previous_trace

And with earlier versions (6.0, 5.0, 4.2) all the above and one more:
1. test_enqueue

**Comment by Ferenc-** [comment]
Hi,

Thank you for the initiative.
We believe that Sidekiq 7.0 support is complete now, so I am closing this for now.
We can reopen if needed.

**Comment by daande** [comment]
@Ferenc- So we have been running on a forked version of this gem that @jeremycw implemented support for Sidekiq 7.0 on.  We recently spent some time trying out the latest release of this gem and can confirm that traces from sidekiq 7.0 are not showing up in instana.  We believe that sidekiq 7.0 is not supported, can you please reopen and investigate?

**Comment by Ferenc-** [comment]
I have created the AHA FR for the full support: https://bigblue.aha.io/ideas/ideas/INSTANA-I-1959

**Pull request closed without merging**